sunpy/extern/modest_image.py

```diff
@@ -0,0 +1,351 @@
+"""
+Modification of Chris Beaumont's mpl-modest-image package to allow the use of
+set_extent.
+"""
+# This file is copied from glue under the terms of the 3 Clause BSD licence. See licenses/GLUE.rst
+
+from __future__ import print_function, division
+
+import matplotlib
+rcParams = matplotlib.rcParams
+
+import matplotlib.image as mi
+import matplotlib.colors as mcolors
+import matplotlib.cbook as cbook
+from matplotlib.transforms import IdentityTransform, Affine2D
+
+import numpy as np
+
+IDENTITY_TRANSFORM = IdentityTransform()
+
+
+class ModestImage(mi.AxesImage):
+
+    """
+    Computationally modest image class.
+
+    ModestImage is an extension of the Matplotlib AxesImage class
+    better suited for the interactive display of larger images. Before
+    drawing, ModestImage resamples the data array based on the screen
+    resolution and view window. This has very little affect on the
+    appearance of the image, but can substantially cut down on
+    computation since calculations of unresolved or clipped pixels
+    are skipped.
+
+    The interface of ModestImage is the same as AxesImage. However, it
+    does not currently support setting the 'extent' property. There
```

Comment:
Change this bit of docstring, since this implementation does allow `extent`?

Reply:
I don't really want to change this file as it's in extern, I just want to be able to copy it out of Glue if there are updates there?

Reply:
yeah, it means we don't have to depend on glue, which is a *large* dependency with a lot of qt and other things, really not worth it.